rp.md

```diff
@@ -0,0 +1,592 @@
+2. Business sends `reservation.modification.response` `kind:9904` message to the customer with `"status":"declined"` to indicate lack of availability for the new time
+3. Customer sends `reservation.response` `kind:9902` message to the business with original time and status `"status":"confirmed"` to maintain original reservation or `"status":"cancelled"` to cancel the original reservation
+4. Message exchange ends
+
+Note: *This flow assumes that there is an existing confirmed reservation initiated by a `reservation.request` `kind:9901` message sent by the customer to the business. All messages should include the `"e"` tag with the rumor ID of the original `reservation.request` message to match the modification to the original reservation.*
+
+### Reservation Cancellation Initated by the Business
+1. Business sends `reservation.response` `kind:9902` message to the customer with `"status":"cancelled"`. Including a note in the `content.message` field is highly encouraged. 
+2. Message exchange ends
+
+No further action is expected from the customer. 
+
+Note: *This flow assumes that there is an existing confirmed reservation initiated by a `reservation.request` `kind:9901` message sent by the customer to the business. All messages should include the `"e"` tag with the rumor ID of the original `reservation.request` message to match the modification to the original reservation.*
+
+### Reservation Cancellation Initated by the Customer
+1. Customer sends `reservation.response` `kind:9902` message to the business with `"status":"cancelled"`. Including a note in the `content.message` field is highly encouraged. 
+2. Message exchange ends
+
+No further action is expected from the business.
+
+Note: *This flow assumes that there is an existing confirmed reservation initiated by a `reservation.request` `kind:9901` message sent by the customer to the business. All messages should include the `"e"` tag with the rumor ID of the original `reservation.request` message to match the modification to the original reservation.*
+
+## JSON Schema Validation
+
+Clients MUST validate payloads against JSON schemas before processing:
+
+- Kind 9901: Validate against `nostrability/schemata/nips/nip-rp/kind-9901/schema.yaml`
+- Kind 9902: Validate against `nostrability/schemata/nips/nip-rp/kind-9902/schema.yaml`
+- Kind 9903: Validate against `nostrability/schemata/nips/nip-rp/kind-9903/schema.yaml`
+- Kind 9904: Validate against `nostrability/schemata/nips/nip-rp/kind-9904/schema.yaml`
+
+
+Invalid payloads MUST be rejected and not processed further.
+
+## Business Discovery 
+
+Businesses MUST advertise their capability to handle reservation messages using [NIP-89](https://github.com/nostr-protocol/nips/blob/master/89.md) Application Handlers.
+
+### Handler Information Event (kind:31990)
+
+Businesses MUST publish a `kind:31990` handler information event that declares support for all reservation message kinds:
```

Comment:
Don't overload 31990, they're already overloaded but this just makes things worse. Handlers are for **apps** that can handle an event kind, not pubkeys. Choose a new kind here.